parser/src/main/java/com/facebook/coresql/warning/DefaultWarningCollector.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.coresql.warning;
+
+import com.facebook.coresql.parser.AstNode;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.LinkedListMultimap;
+import com.google.common.collect.ListMultimap;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public class DefaultWarningCollector
+        implements WarningCollector
+{
+    private final ListMultimap<WarningCode, CoreSqlWarning> warnings = LinkedListMultimap.create();
```

Comment:
Use Multimap instead of ListMultimap, if you can. It's always better to go with the most generic interface.

Reply:
Done